### PR TITLE
Add golang tests

### PR DESCRIPTION
**Pull request by sporksmith**
This adds some simple golang tests, most of which currently fail (and are disabled) under Shadow.

Added an `--extra` flag to the `build` and `test` subcommands of the `setup` script. Including this command in both now builds and runs the golang tests.

Enabled these tests in the CI.

### CODE REVIEW

File: src/test/golang/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    CONFIGURATIONS extra
+    PROPERTIES
+      LABELS golang)
+add_shadow_tests(
+    BASENAME gc
+    # BUG: https://github.com/shadow/shadow/issues/1931
+    SKIP_METHODS preload
+    CONFIGURATIONS extra 
+    PROPERTIES
+      LABELS golang)
```
Comment (stevenengler) [inline src/test/golang/CMakeLists.txt:52]: Not sure if it's an issue, but one instance of `add_shadow_tests()` has `LABELS` under `PROPERTIES` and the other doesn't.

**Review by stevenengler — APPROVED** [review]

File: setup [hunk 2]
```diff
@@ -259,6 +275,9 @@ def test(args, remaining):
     if args.testname_regex is not None:
         testcmd += " -R {}".format(args.testname_regex)
 
+    if args.do_extra_test:
+        testcmd += " -C extra"
```
Comment (stevenengler) [inline setup:279]: Maybe also update the documentation that uses `-C extra` and change it to use this new flag instead:

https://shadow.github.io/docs/guide/developer_guide.html?highlight=extra#extra-tests  
https://shadow.github.io/docs/guide/getting_started_tor.html?highlight=extra#getting-started-tor

File: setup [hunk 3]
```diff
@@ -220,22 +227,31 @@ def build(args, remaining):
 
     # call cmake to configure the make process, wait for completion
     logging.info("running \'{0}\' from \'{1}\'".format(cmake_cmd, os.getcwd()))
-    retcode = subprocess.call(cmake_cmd.strip().split())
+    retcode = subprocess.call(cmake_cmd.strip().split(), cwd=builddir)
```
[the remark this answers is not in the document]
Comment (sporksmith) [reply setup]: Done

File: setup [hunk 4]
```diff
@@ -220,22 +227,31 @@ def build(args, remaining):
...
+        return retcode
+
+    # call make, wait for it to finish
+    make = f"make -j{args.njobs}"
+    logging.info("calling " + make)
+    retcode = subprocess.call(shlex.split(make), cwd=builddir)
+    logging.debug("make returned " + str(retcode))
+    if retcode != 0:
+        logging.error(f"Non-zero return code {retcode} from cmake.")
+        return retcode
```
[the remark this answers is not in the document]
Comment (sporksmith) [reply setup]: Done

File: setup [hunk 5]
```diff
@@ -220,22 +227,31 @@ def build(args, remaining):
...
         logging.info("calling " + make)
-        retcode = subprocess.call(shlex.split(make))
+        retcode = subprocess.call(shlex.split(make), cwd=builddir)
         logging.debug("make returned " + str(retcode))
-        if retcode == 0: logging.info("build successful!")
-        else: logging.error("Non-zero return code from make.")
-    else: logging.error(" Non-zero return code from cmake.")
+        if retcode != 0:
+            logging.error(f"Non-zero return code {retcode} from cmake.")
+            return retcode
```
[the remark this answers is not in the document]
Comment (sporksmith) [reply setup]: Done

File: setup (hunk 2 above)
Comment (sporksmith) [reply setup:279]: Good catch; done

File: src/test/golang/CMakeLists.txt (hunk 1 above)
Comment (sporksmith) [reply src/test/golang/CMakeLists.txt]: Thanks, good catch. It belongs under `PROPERTIES`; I think in the instance where I forgot to put it there, it was getting interpreted as additional `CONFIGURATIONS`

**Pull request merged**